### PR TITLE
Use the built JS, and run some python to fix it up

### PR DESCRIPTION
**Pull request by dannymandel**
@datadavev this is what we discussed this morning.  It's doing something funny that I can't explain on my local machine.  Going to push it out to mars to give it a whirl there…

### CODE REVIEW

File: isb/Dockerfile [hunk 1]
```diff
@@ -1,26 +1,4 @@
 # syntax=docker/dockerfile:1
-
```
Comment (dannymandel) [inline isb/Dockerfile:2]: No longer running the build during the image building process

File: isb/Dockerfile [hunk 2]
```diff
@@ -56,10 +34,13 @@ RUN mkdir /var/log/isamples
...
 # Put the React build output inside isb_web ui directory for serving as part of the uvicorn installation
-# This statement is a bit confusing -- the --from=node_build is the build stage we are copying *from*,
-# and the /app/build is the path in *that* image (the multistage builds produce separate images).  The
-# dest path (/app/isb_web/ui) is the path in *this* image.
-COPY --from=node_build /app/build/ /app/isb_web/ui/
+# This is built as part of the web_ui github action -- then the output is checked out in the gh-pages branch, copied to 
+# the container, and we run python inside the container to search and replace the necessary values in the container itself.
+ARG ISB_SITEMAP_PREFIX=https://mars.cyverse.org
+RUN echo "ISB_SITEMAP_PREFIX IS $ISB_SITEMAP_PREFIX"
+COPY ./isamples_webui/build/ /app/isb_web/ui/
```
Comment (dannymandel) [inline isb/Dockerfile:41]: Instead, take the build output that's been checked in on the `gh-pages` branch (as part of our pushes to develop), and copy it into the container here.

File: isb/Dockerfile [hunk 3]
```diff
@@ -56,10 +34,13 @@ RUN mkdir /var/log/isamples
...
+COPY ./massage_webui_output.py .
+RUN python /app/massage_webui_output.py -u https://hyde.cyverse.org/isamples_central -r $ISB_SITEMAP_PREFIX -p /app/isb_web/ui/static/js
```
Comment (dannymandel) [inline isb/Dockerfile:43]: Then run the python script in the container to fix up the source.  This appears to be running locally for me, but isn't producing the output I expect.

**Comment by dannymandel** [comment]
Need to merge this one to test it out on mars…feel free to leave feedback and I will address in a followup commit.

**Comment by dannymandel** [comment]
See also: https://github.com/isamplesorg/isamples-ansible/commit/10b04afdb4228f237535e77d84b7ad0388b829c2

**Comment by datadavev** [comment]
There should be no need to massage the built js app. I think the fundamental problem is that `config.js` for the web ui is included in the build and is not settable on application initialization.

I think the config should include sensible defaults, but with an option to set properties at app startup. Those properties should be settable by an optional `local_config.json` available at a well known location relative to the app (e.g. "."). The local_config would be created for the respective environment.

One approach for achieving may be to replace the config dict with a class that loads from local_config on instantiation.

**Pull request merged**